### DISCUSSION TITLE
Limit the available distributions

### DISCUSSION THREAD

**Discussion by mpriem** [microsoft/WSL#12050] [Q&A] [answered]
We want to provide WSL in a highly regulated environment and are looking for a way to limit the available distributions. I wasn't able to find a way to just allow our own modified image to be available for the developer.

Currently everyone can just install openSUSE or kali-linux and ignore our custom image that has certain security software on it.

Has anyone found a good approach for this?

**Comment by zcobol** [comment]
If it's a `highly regulated environment` why would your users install anything but what's required by your ...highly regulated environment? You could setup a local `winget` repo to host your `own modified image` and nothing else.

**Reply by mpriem** [reply]
That is why I'm asking... Don't need to be a dick about it.

**Comment by OneBlue** [comment, marked as answer]
@mpriem: You can now do that by using a [custom distribution manifest](https://learn.microsoft.com/en-us/windows/wsl/build-custom-distro#adding-your-distribution-to-wsl---install-for-your-enterprise-or-business-group).

In your specific case, since you want to limit the available distributions, `DistributionListUrl` is the right key to set.

**Reply by jamesdboone** [reply]
When/how does this manifest get refreshed?  Seems like it works great, but changes to the manifest aren't getting picked up...
